archive_updater.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import os
+
+# TODO(willliu@google.com): bump this value up
+MAX_NUMBER_OF_RECENT_COMMITS = 2
+
+def fill_message_directory(archive_path: str, directory_path: str, last_used_commit_hash: str) -> str:
+    '''Given a git repo path, directory path, and last used commit hash, this method updates the git repo,
+    then retrieves the MAX_NUMBER_OF_RECENT_COMMITS recent commits and converts them into files stored in
+    the directory corresponding to the passed in directory path.
+
+    Parameters:
+    archive_path - the path to the email archive
+    directory_path - the directory where files will be temporarily stored
+    last_used_commit_hash - the last commit hash which will be used as the ending point of this function
+
+    Returns the most recent commit hash that was processed in this method call
+
+    Throws CalledProcessError and Exception'''
+
+    fetch = subprocess.check_call(['git', '-C', archive_path, 'fetch'])
+
+    output = subprocess.check_output(
+        ['git', '-C', archive_path, 'log', '--format=format:%H', '-n',str(MAX_NUMBER_OF_RECENT_COMMITS)])
+    message_hashes = output.decode('utf-8').split()
+
+    if len(message_hashes) == 0:
+        raise Exception('There are no recent commits on the given repo')
+
+    for i, hash in enumerate(message_hashes):
```

Comment:
We don't need `i` anymore, can do
```suggestion
    for hash in message_hashes:
```